packages/extension/src/serviceWorker/attachCDP.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * This function will attach the debugger to the given target.
+ * @param {{ [key: string]: number | string }} target The target where debugger needs to be attached.
+ */
+export default async function attachCDP(target: {
+  [key: string]: number | string;
+}) {
+  try {
+    await chrome.debugger.attach(target, '1.3');
+    await chrome.debugger.sendCommand(target, 'Target.setAutoAttach', {
+      autoAttach: true,
+      waitForDebuggerOnStart: false,
+      flatten: true,
```

Comment:
Please add some comment on what `flatten: true,` does. If it attaches CDP debugger to all sub-frames to single level.